library/general/src/lib/ui/password_dialog.rb

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "shellwords"
+
+require "ui/dialog"
+require "yast2/popup"
+
+Yast.import "UI"
+Yast.import "Label"
+
+module UI
+  # Dialog that asks for password. It returns String password or nil if aborted.
+  #
+  class PasswordDialog < UI::Dialog
+    extend Yast::I18n
+    extend Yast::UIShortcuts
+
+    # @param label [String] intention of password e.g. "Encrypted autoyast profile."
+    # @param confirm [Boolean] if double password entry is required.
+    #   Usually when new password is specified.
+    def initialize(label, confirm: false)
+      textdomain "autoinst"
+      @confirm = confirm
+      @label = label
+
+      super()
+    end
+
+    def dialog_content
+      res = VBox(
+        Left(Heading(@label)),
+        HBox(
+          MinWidth(18, Label(_("Password:"))),
```

Comment:
We already have [Label.Password](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Label.rb#L321) and [Label.ConfirmPassword](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Label.rb#L327), used like [here](https://github.com/yast/yast-users/blob/master/src/include/users/dialogs.rb#L517-L518).